Automattic/wp-calypso
# Update FSE template part hover styles

### Summary

This PR changes some of the functionality and design of the template part hover actions to make it less jarring. 

![Updated-FSE-header-hover](https://user-images.githubusercontent.com/8892849/62653099-d718c300-b92a-11e9-8230-4954514c5e69.gif)

<hr>

### Specifics

* Old hover styles removed. New hover styles are a gray left border and allowing the breadcrumb to be visible.
* Removed the functionality to disable selecting template blocks. Instead, duplicated it inside of the template block component to ensure that the sidebar is closed if the block is selected/unselect the block when the sidebar is opened.
* Placeholder is only visible when the block is selected.

<hr>

### Screenshots

**Hover state**

<img width="687" alt="Screen Shot 2019-08-07 at 1 34 47 PM" src="https://user-images.githubusercontent.com/8892849/62653175-fca5cc80-b92a-11e9-8f1d-e18173cd8046.png">

**Focus state**

<img width="697" alt="Screen Shot 2019-08-07 at 1 34 54 PM" src="https://user-images.githubusercontent.com/8892849/62653184-03ccda80-b92b-11e9-9d76-e3f1ae76fb0a.png">

<hr>

### Testing

* Setup FSE using your local development environment of choice.
* Open up the new page editor. 
* Ensure that the hover styles are visible when hovering a template block, but not for the post content slot.
* Click on a template part block. Ensure that the placeholder is visible.
* Ensure that it is not possible to have the placeholder visible and the settings sidebar open at the same time.

Fixes #34659
